### ISSUE TITLE
Rule Proposal: `vue/no-restricted-component-names`

### ISSUE THREAD

**Issue by qmhc** [vuejs/eslint-plugin-vue#2182]
**Please describe what the rule should do:**

Restrict some component names via a list of pattern.

**What category should the rule belong to?**

[ ] Enforces code style (layout)
[x] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

Options:

```json
{
  "vue/no-restricted-component-names": ["error", "Foo", "/^Baz(One|Two)/"]
}
```

```vue
<script>
<!-- ✗ BAD -->
export default {
  name: 'Foo'
}

<!-- ✗ BAD -->
export default {
  name: 'BazTwo'
}

<!-- ✓ GOOD-->
export default {
  name: 'Allow'
}
</script>
```

**Additional context**

Related https://github.com/vuejs/eslint-plugin-vue/issues/2180#issuecomment-1551298495.

**Comment by ItMaga** [comment]
I think there can be cases when we need more detailed options, for example, when the team has some convention for suggestions or a custom error message. What do you think about this?

As a variant:

```js
string[]
// or
{
  pattern: "string or regex",
  suggestion: "string", // optional
  message: "string" // optional
 }
```
